File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program118.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and copy that characters of that string into 
//		another string by converting all small characters into capital case.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{
	private:
		char *src,*dest;

	public:
		Solution(char *src,char *dest)
		{
			this->src=src;
			this->dest=dest;
		}

		
		void StrCpyCap()
		{

			while(*src != '\0')
			{
				if(*src>='a' && *src<='z')
				{	
					*dest = (*src)-32;
				}
				else if(*src>='A' && *src<='Z')
				{
					*dest=*src;
				}
				else
				{
					*dest = *src;
				}
				dest++;
				src++;
			}
			*dest='\0';

		}
};


int main()
{
	char arr[30] = "Vishal Patil 2";
	char brr[30];

	Solution *sObj = new Solution(arr,brr);

	sObj->StrCpyCap();
	cout<<brr<<"\n";

	return 0;
}

/*
OUTPUT:

VISHAL PATIL 2

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program118.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write java program which accept N numbers from user and accept one another number as NO ,
//		 return index of last occurrence of that NO.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class ArrayX
{
	public int iSize,Brr[],iNo;

	public ArrayX(int iSize,int iNo)
	{
		Brr = new int[iSize];
		this.iNo=iNo;
	}

	public void Accept()
	{
		int iCnt=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter elements in Array: ");
		for(iCnt=0;iCnt<Brr.length;iCnt++)
		{
			Brr[iCnt] = sObj.nextInt();
		}
	}
}

class Number extends ArrayX
{

	public Number(int iValue,int iNo)
	{
		super(iValue,iNo);
	}

	public int LastOcc()
	{
		int iCnt=0,iLast=-1;
		for(iCnt=Brr.length-1;iCnt>=0;iCnt--)
		{
			if(Brr[iCnt]==iNo)
			{
				iLast=iCnt;
				break;	
			}
		}

		return iLast;
	}		
}

class Program3

{
	public static void main(String arg[])
	{
		int iCnt=0, iRet=0;

		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter number of elements: ");
		int iSize = sObj.nextInt();

		System.out.println("Enter the elements: ");
		int iNo = sObj.nextInt();

		Number nObj = new Number(iSize,iNo);
		nObj.Accept();
		iRet = nObj.LastOcc();

		System.out.println("Output:"+iRet);
	}
}

/*
OUTPUT:

Enter number of elements: 
6
Enter the elements: 
66
Enter elements in Array: 
85
66
3
66
93
88
Output:3

*/
